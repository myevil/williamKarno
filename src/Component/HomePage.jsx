import React, { Component } from 'react'
import { getFetch } from '../Tools/api';
import '../Scss/HomePage.scss';
import ListProduct from './Product/ListProduct';
import ShoppingCart from './ShoppingCart/ShoppingCart';

export class HomePage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      data: null,
      shoppingCartData: [],
    }
  }

  componentDidMount() {
    getFetch('http://s3.irvinsaltedegg.com.s3-ap-southeast-1.amazonaws.com/engineering-test/products.json')
      .then(resp => {
        this.setState({
          data: resp.data,
        })
      })
  }

  render() {
    const {data, shoppingCartData} = this.state;
    return (
      <div className='scss-homepage-container'>
        <div className='scss-product-container'>
          <ListProduct
            data={data}
          />
        </div>
        <hr />
        {data && <div>
          <table className='scss-shopping-cart'>
            <thead>
              <tr>
                <th>&nbsp;</th>
                <th>product</th>
                <th>price</th>
                <th>quantity</th>
                <th>total</th>
              </tr>
            </thead>
            <tbody>
              <ShoppingCart
                ShoppingCart={shoppingCartData}
              />
            </tbody>
          </table>
        </div>}
      </div>
    )
  }
}

export default HomePage
