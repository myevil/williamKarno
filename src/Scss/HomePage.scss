.scss-homepage-container {
  margin: 20px;
}
.scss-product-container {
  display: flex;
  flex-direction: row;
}
.scss-grid-md-3 {
  display: flex;
  align-items: center;
  flex-direction: column;
  width: 33%;
  margin: 0px 10px;
}
.scss-grid-md-6 {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 66%;
  margin: 0px 10px;
}
