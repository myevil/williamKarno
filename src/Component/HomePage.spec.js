import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import HomePage from './HomePage';

configure({ adapter: new Adapter() });

describe('test hompage', () => {
  const wrapper = shallow(<HomePage />)
  it('default state at homepage must be have', () => {
    expect(wrapper.state().data).toBe(null);
  })
  it('after component did mount state should contain data', () => {
    
  })
})
