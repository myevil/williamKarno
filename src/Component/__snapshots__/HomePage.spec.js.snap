// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test hompage should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomePage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="scss-product-container"
        >
          <ListProduct
            data={null}
          />
        </div>,
        <hr />,
        null,
      ],
      "className": "scss-homepage-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ListProduct
            data={null}
          />,
          "className": "scss-product-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="scss-product-container"
          >
            <ListProduct
              data={null}
            />
          </div>,
          <hr />,
          null,
        ],
        "className": "scss-homepage-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ListProduct
              data={null}
            />,
            "className": "scss-product-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
